spec/tapioca/cli/clean_shims_spec.rb

```diff
@@ -0,0 +1,261 @@
+# typed: true
+# frozen_string_literal: true
+
+require "spec_with_project"
+
+module Tapioca
+  class CleanShimsTest < SpecWithProject
```

Comment:
Can we add an example for a shim that only gets partially deleted? Something like
```ruby
# shim
# Both methods are generated by Tapioca, but because one has a signature, we only deleted `baz`
class Foo
  sig { params(arg: Integer).void }
  def bar(arg); end

  def baz; end
end

# generated
class Foo
  def bar(arg); end
  def baz; end
end

# Which should then produce the following shim after cleanup
class Foo
  sig { params(arg: Integer).void }
  def bar(arg); end
end
```